### PR TITLE
#15337: Fix incorrectly sized cb in remote cb microbenchmark

### PR DESCRIPTION
**Pull request by tt-aho**
### Ticket
https://github.com/tenstorrent/tt-metal/issues/15510

### Problem description
Output cb was wrong sized and exceeds actual L1 buffer size on a bank for remote cb test.
Problem with validation if user/op plans to update both the dynamic buffer and the total cb size, in that updating only one individually would compare to the old value of the other and potentially fail assertion.

### What's changed
Fix size calculation in test.
Add a new api that updates both the global buffer and the total size at the same time, so that it can perform validation on both new values.

### Checklist
- [x] Post commit CI passes
- [ ] Blackhole Post commit (if applicable)
- [ ] Model regression CI testing passes (if applicable)
- [ ] Device performance regression CI testing passes (if applicable)
- [ ] New/Existing tests provide coverage for changes

### CODE REVIEW

File: tt_metal/host_api.hpp
Comment (bbradelTT) [inline tt_metal/host_api.hpp:273]: Please update the doc string for the other method to indicate to use this one if you need to set both and/or provide an ordering for what order to call the method in.

**Review by bbradelTT — APPROVED** [review]

**Review by yugaoTT — APPROVED** [review]

**Review by ntarafdar — APPROVED** [review]

File: ttnn/cpp/ttnn/operations/eltwise/binary/device/broadcast_height_multi_core_sharded_optimized_program_factory.cpp
Comment (tt-aho) [inline ttnn/cpp/ttnn/operations/eltwise/binary/device/broadcast_height_multi_core_sharded_optimized_program_factory.cpp:92]: @eyonland not directly related to this issue but this calculation for shard size doesn't seem correct for data types like bfp8. I saw we assert that a is tile layout so bfp8 is a valid dtype, and element_size() returns 1 for bfp8. Ex if the shard was 1 tile this would calculate 1024B, but bfp8 tile size is 1088B.

**Review by aliuTT — APPROVED** [review]

**Comment by tt-rkim** [comment]
Force-merged to unblock https://github.com/tenstorrent/tt-metal/issues/15510

IT'S GIVING THANKSGIVING

**Comment by tt-aho** [comment]
@yan-zaretskiy @eyonland @patrickroberts as mentioned above we force merged without your approval to unblock a P0 issue. Please leave any comments/feedback regarding these changes when you're available and I will address them.

**Pull request merged**
